language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/WhereClauseNodeContext.java

```diff
@@ -57,12 +58,21 @@ public WhereClauseNodeContext() {
             completionItems.addAll(this.expressionCompletions(context));
         }
         this.sort(context, node, completionItems);
-        
+
         return completionItems;
     }
 
     @Override
     public boolean onPreValidation(BallerinaCompletionContext context, WhereClauseNode node) {
         return !node.whereKeyword().isMissing();
     }
+
+    private boolean cursorAtTheEndOfExpression(BallerinaCompletionContext context, WhereClauseNode node) {
```

Comment:
Shall we move this method to `IntermediateClauseNodeContext`? If an extending context needs to have a specific logic, then the particular context can override this. WDYT?

Reply:
I thought the same. But the issue is, `IntermediateClauseNode` doesn't have the `expression()` method. If we move this to parent class, we will have to do `instanceof` checks for each sub type.

Reply:
If we manipulate only the expression what if we pass the expression as an argument?

Reply:
Refactored to get the expression from child classes and do the above check within `IntermediateClauseNodeContext`.